### PR TITLE
Compilation: strip debug info from ReleaseSmall by default

### LINKED ISSUE #6676: strip debug info from ReleaseSmall builds by default
Repro code:
```zig
const std = @import("std");
const linux = std.os.linux;

pub fn exit() void {
    _ = linux.syscall1(.exit, 0);
}

pub fn main() void {
    exit();
}
```

Running `zig build-exe main.zig -target arm-linux-none -O ReleaseSmall` for arm produces an executable that is 707KB in size. However, running `zig build-exe main.zig -target arm-linux-none -O ReleaseSmall` for i386 produces an executable that is just 61KB.

Appending `--strip` "fixes" the issue as the executables become 5KB and 3KB.

### PR DESCRIPTION
**Pull request by alichraghi**
Closes #6676
```
ali@linux ~/d/zig (o0)> build/zig2 build-exe z.zig -O ReleaseSmall
ali@linux ~/d/zig (o0)> ls -lha ./z ./z.o
-rwxr-xr-x 1 ali ali 8.5K Oct  4 17:52 ./z*
-rw-r--r-- 1 ali ali  12K Oct  4 17:52 ./z.o
ali@linux ~/d/zig (o0)> build/zig2 build-exe z.zig -O ReleaseSmall --strip
ali@linux ~/d/zig (o0)> ls -lha ./z ./z.o
-rwxr-xr-x 1 ali ali 8.5K Oct  4 17:52 ./z*
-rw-r--r-- 1 ali ali  12K Oct  4 17:52 ./z.o
ali@linux ~/d/zig (o0)> build/zig2 build-exe z.zig -O ReleaseSmall --no-strip
ali@linux ~/d/zig (o0)> ls -lha ./z ./z.o
-rwxr-xr-x 1 ali ali 426K Oct  4 17:53 ./z*
-rw-r--r-- 1 ali ali 729K Oct  4 17:53 ./z.o
ali@linux ~/d/zig (o0)> build/zig2 build-exe z.zig
ali@linux ~/d/zig (o0)> ls -lha ./z ./z.o
-rwxr-xr-x 1 ali ali 976K Oct  4 17:53 ./z*
-rw-r--r-- 1 ali ali 1.4M Oct  4 17:53 ./z.o
```

### CODE REVIEW

File: src/main.zig [hunk 1]
```diff
@@ -404,7 +404,8 @@ const usage_build_generic =
     \\  -fno-builtin              Disable implicit builtin knowledge of functions
     \\  -ffunction-sections       Places each function in a separate section
     \\  -fno-function-sections    All functions go into same section
-    \\  --strip                   Omit debug symbols
+    \\  --strip                   Force-enable debug symbols
+    \\  --no-strip                Force-disable debug symbols
```
Comment (r00ster91) [inline src/main.zig:408]: These descriptions are confusing to me. `--strip` force-enables debug symbols? It omits them so it rather "disables" them. I think this is better word choice:
```suggestion
    \\  --strip                   Omit debug symbols
    \\  --no-strip                Keep debug symbols
```

**Comment by r00ster91** [comment]
Also, Andrew suggested `-fstrip` and `-fno-strip` instead of `--strip` and `--no-strip` (https://github.com/ziglang/zig/issues/6676#issuecomment-710764280); have you considered that?

**Comment by alichraghi** [comment]
thanks. i missed that

**Comment by alichraghi** [comment]
Now, would it make sense to disable `strip` by default for `zig test` or `zig build test` (or both)?
\* by "default", i mean it's `null` so it get enabled in **ReleaseSmall**

**Comment by Vexu** [comment]
Whoops, looks like the Windows failure wasn't spurious but caused by a need for a new tarball (sigh). #13055 also needs one so maybe these two should be handled together.

**Comment by topolarity** [comment]
Looks like this also broke `test/link/wasm/producers`, which was merged a few days ago (after the tests had passed in this PR)

**Comment by Luukdegram** [comment]
Looks like this also caused the CI to error out for the linker test for the 'producers' section.
Fixed this in 8635c18e4091bd3034121ca98439bf40bce72324

**Pull request merged**